src/iam/iam-base.ts

```diff
@@ -523,6 +523,27 @@ export class IAMBase {
     throw new MethodNotAvailableInNodeEnvError("deleteSubdomain");
   }
 
+  protected async deleteDomain({ namespace }: { namespace: string }) {
+    if (this._signer && this._ensRegistry && this._ensResolver) {
+      const ensRegistryWithSigner = this._ensRegistry.connect(this._signer);
+      const hash = namehash(namespace);
+      const ttl = bigNumberify(0);
+
+      const tx = await ensRegistryWithSigner.setRecord(
+        hash,
+        emptyAddress,
+        emptyAddress,
+        ttl,
+        this._transactionOverrides
+      );
+
+      await tx.wait();
+
+      return;
+    }
+    throw new MethodNotAvailableInNodeEnvError("deleteSubdomain");
```

Comment:
Just curious, why is this method not available in node?

Reply:
Yeah, good point 👍